### PR TITLE
[repo] Refactor CI to use Component.proj

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Refactor `ci.yml` & `Component.BuildTest.yml` workflows to utilize `Component.proj` in order to reduce the number of projects which need to be maintained in the repo

## Details

#1808 introduced `Component.proj` which can be used to restore\build\test\pack components which have more or less a standard/simple project structure. Using it in the CI workflows allows us to shed a bunch project files which were more or less boilerplate.

### CODE REVIEW

**Comment by Kielek** [comment]
Note for myself, standard layout:

* `/src/OpenTelemetry.{PackageName}/OpenTelemetry.{PackageName}.csproj`
* `/test/OpenTelemetry.{PackageName}.Tests/OpenTelemetry.{PackageName}.Tests.csproj`
* `/test/OpenTelemetry.{PackageName}.Benchmark/OpenTelemetry.{PackageName}.Benchmark.csproj` - potentially can be unified with the next entry
* `/test/OpenTelemetry.{PackageName}.Benchmarks/OpenTelemetry.{PackageName}.Benchmarks.csproj`
* `/test/OpenTelemetry.{PackageName}.Stress/OpenTelemetry.{PackageName}.Stress.csproj`

File: .github/workflows/ci-Exporter.OneCollector-Integration.yml [hunk 1]
```diff
@@ -46,13 +46,13 @@ jobs:
...
+      run: dotnet restore build/Projects/Component.proj
 
-    - name: dotnet build build/Projects/${{env.PROJECT}}.proj
-      run: dotnet build build/Projects/${{env.PROJECT}}.proj --configuration Release --no-restore
+    - name: dotnet build Component.proj[${{ env.BUILD_COMPONENT }}]
+      run: dotnet build build/Projects/Component.proj --configuration Release --no-restore
 
-    - name: dotnet test test/${{env.PROJECT}}.Tests
-      run: dotnet test test/${{env.PROJECT}}.Tests --filter CategoryName=OneCollectorIntegrationTests --framework ${{ matrix.version }} --configuration Release --no-restore --no-build --logger:"console;verbosity=detailed"
+    - name: dotnet test Component.proj[${{ env.BUILD_COMPONENT }}]
```
Comment (Kielek) [inline .github/workflows/ci-Exporter.OneCollector-Integration.yml:55]: I do not fully understand the syntax here, first time I see `[` `]` in this context. Do you have any documentation with explanation what is the result of call
`Component.proj[${{ env.BUILD_COMPONENT }}]`?

Comment (CodeBlanch) [reply .github/workflows/ci-Exporter.OneCollector-Integration.yml]: @Kielek 

I refactored things to try and make it more clear what is going on. Check out this commit: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1811/commits/77191cab2bd582d63b384029508fe7bf1c19a263

LMK if you feel it needs more work.

I'm happy to explain here but I figured making the code more self-explanatory would be better in the long run.

Comment (Kielek) [reply .github/workflows/ci-Exporter.OneCollector-Integration.yml]: Nice! Now it is clear to me.

**Review by Kielek — APPROVED** [review]
Changes looks good.

Before merging, please check your commits added to this PR. A lot of them are assigned to github-actions[bot] account. I suppose it is not true.

**Comment by CodeBlanch** [comment]
@Kielek 

> Before merging, please check your commits added to this PR. A lot of them are assigned to github-actions[bot] account. I suppose it is not true.

Ya sorry about that. I was testing one of the scripts locally and it switched my repo to use the bot info for git config. I fixed it 🤣

**Comment by CodeBlanch** [comment]
@Kielek 

> Note for myself, standard layout:

Ya some projects use "Benchmarks" and others uses "Benchmark" if we clean that up we could simplify `Component.proj`. There are also some instrumentation projects with example apps forcing them to have dedicated `.proj` files. If we could choose a naming convention for the example apps which is more deterministic we could include those in `Component.proj` as well and drop a few more of the dedicated projects.

**Pull request merged**